code/modules/security_levels/keycard_authentication.dm

```diff
@@ -142,23 +142,15 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/keycard_auth, 26)
 
 GLOBAL_VAR_INIT(emergency_access, FALSE)
 /proc/make_maint_all_access()
-	for(var/area/maintenance/M as anything in get_areas(/area/maintenance, SSmapping.levels_by_trait(ZTRAIT_STATION)[1]))
-		for(var/turf/in_area as anything in M.get_contained_turfs())
-			for(var/obj/machinery/door/airlock/A in in_area)
-				A.emergency = TRUE
-				A.update_icon()
-				A.wires.ui_update()
+	for(var/obj/machinery/door/airlock/A)
+		A.set_door_to_all_access()
```

Comment:
Don't use for loops that have no in clause, this is a loop over every atom in the world which is very slow. Also don't use `in area` since that is also a loop over every atom in the world. Use .get_contained_turfs() on an area instead.

Reply:
If you want to make a good improvement on the code, add a global signal and register to it on doors that need to update to significantly cut down on the things that you are doing when it happens. Or you could also loop through airlocks in GLOB.machines, but that has a lot of waste but is easier